Calendar.java

```diff
@@ -2,64 +2,64 @@
 import java.util.concurrent.TimeUnit;
 
 public class Calendar {
-	
-	private static Calendar SeLf;
-	private static java.util.Calendar CaLeNdAr;
-	
-	
-	private Calendar() {
-		CaLeNdAr = java.util.Calendar.getInstance();
-	}
-	
-	public static Calendar INSTANCE() {
-		if (SeLf == null) {
-			SeLf = new Calendar();
-		}
-		return SeLf;
-	}
-	
-	public void incrementDate(int days) {
-		CaLeNdAr.add(java.util.Calendar.DATE, days);		
-	}
-	
-	public synchronized void Set_dATE(Date date) {
-		try {
-			CaLeNdAr.setTime(date);
-	        CaLeNdAr.set(java.util.Calendar.HOUR_OF_DAY, 0);  
-	        CaLeNdAr.set(java.util.Calendar.MINUTE, 0);  
-	        CaLeNdAr.set(java.util.Calendar.SECOND, 0);  
-	        CaLeNdAr.set(java.util.Calendar.MILLISECOND, 0);
-		}
-		catch (Exception e) {
-			throw new RuntimeException(e);
-		}	
-	}
-	public synchronized Date Date() {
-		try {
-	        CaLeNdAr.set(java.util.Calendar.HOUR_OF_DAY, 0);  
-	        CaLeNdAr.set(java.util.Calendar.MINUTE, 0);  
-	        CaLeNdAr.set(java.util.Calendar.SECOND, 0);  
-	        CaLeNdAr.set(java.util.Calendar.MILLISECOND, 0);
-			return CaLeNdAr.getTime();
-		}
-		catch (Exception e) {
-			throw new RuntimeException(e);
-		}	
-	}
 
-	public synchronized Date Due_Date(int loanPeriod) {
-		Date NoW = Date();
-		CaLeNdAr.add(java.util.Calendar.DATE, loanPeriod);
-		Date DuEdAtE = CaLeNdAr.getTime();
-		CaLeNdAr.setTime(NoW);
-		return DuEdAtE;
-	}
-	
-	public synchronized long Get_Days_Difference(Date targetDate) {
-		
-		long Diff_Millis = Date().getTime() - targetDate.getTime();
-	    long Diff_Days = TimeUnit.DAYS.convert(Diff_Millis, TimeUnit.MILLISECONDS);
-	    return Diff_Days;
-	}
+    private static Calendar thisInstance;
+    private static java.util.Calendar javaCalendar;
+
+
+    private Calendar() {
+        javaCalendar = java.util.Calendar.getInstance();
+    }
+
+    public static Calendar getInstance() {
+        if (thisInstance == null) {

```

Comment:
D4: Missing "this" qualifier for thisInstance. Suggest changing "thisInstance" variable name to avoid "this.thisInstance". "self" is actually an acceptable name.

Reply:
renamed thisInstance -> self
'this.' qualifier not applicable in this context